public/app/features/query/state/DashboardQueryRunner/DashboardQueryRunner.ts

```diff
@@ -29,7 +31,7 @@ class DashboardQueryRunnerImpl implements DashboardQueryRunner {
     private readonly dashboard: DashboardModel,
     private readonly timeSrv: TimeSrv = getTimeSrv(),
     private readonly workers: DashboardQueryRunnerWorker[] = [
-      new AlertStatesWorker(),
+      config.featureToggles.ngalert ? new UnifiedAlertStatesWorker() : new AlertStatesWorker(),
```

Comment:
nit: you could also move feature toggle check to the alert state workers `canWork` function. But this works too!

Reply:
I thought the current solution is more obvious , but no strong feelings :) do you prefer checking i n`canWork`?

Reply:
No worries, no strong opinion, keep as is will do!